core/schema/modulesource.go

```diff
@@ -149,6 +149,10 @@ func (s *moduleSourceSchema) Install() {
 			ArgDoc("generator", `The generator to use`).
 			ArgDoc("outputDir", `The output directory for the generated client.`).
 			ArgDoc("dev", `Generate in developer mode`),
+
+		dagql.NodeFunc("introspectionJSONFile", s.introspectionJSONFile).
```

Comment:
It makes some sense to have this in `ModuleSource` but it would also be helpful to be able to get the core API bindings. Right now some SDKs are getting this from the `codegen` binary when updating the core bindings in the client library. This would make that much more accessible.

So what about a top-level field (or maybe an `introspection` field returning an `Introspection` type, to scope introspection related APIs in the future, the same way we have `engine`) that receives an optional ModuleSource instead? Then if not provided, you still get the core schema. If provided, you additionally get a module's.

Reply:
For example, the Python SDK generates core bindings using this function:

https://github.com/dagger/dagger/blob/c38d8440dc933483ca51ea7b8a54463d191844df/sdk/python/dev/src/python_sdk_dev/main.py#L132-L138

And it gets the introspection file through the root module:

https://github.com/dagger/dagger/blob/c38d8440dc933483ca51ea7b8a54463d191844df/.dagger/sdk_python.go#L125

Which in turn gets it from the `codegen` CLI:

https://github.com/dagger/dagger/blob/c38d8440dc933483ca51ea7b8a54463d191844df/.dagger/sdk.go#L98

But, this means that I can't use the dev module directly to generate those bindings. Getting the introspection from the API would free that restriction.

Reply:
Why don't we just send the module source to the codegen? Because we would need it as argument to `introspection` in the coreAPI binding so that's the same problem no?

Reply:
No, because the core API doesn't have a ModuleSource, right? Unless you instantiate one with an invalid value on purpose and tell it not to fail, then you'd get the core functions by default?

Reply:
Ohhh I see! okay yeah